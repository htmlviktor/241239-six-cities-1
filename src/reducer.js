import {loadOffers} from './actions';

const initialState = {
  currentCity: `Amsterdam`,
  listCities: new Set([...data.map((offer) => offer.city)]),
  offers: []
};

const Operation = {
  loadOffers: () => (dispatch, _getState, api) => {
    return api.get(`/hotels`)
      .then((response) => {
        dispatch(loadOffers(response.data));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `CITY_CHANGE`:
      return Object.assign({}, state, {
        currentCity: action.city
      });
    case `LOAD_OFFERS`:
      return Object.assign({}, state, {
        offers: action.offers
      });
    default: return state;
  }
};


export {
  reducer,
  Operation
};
