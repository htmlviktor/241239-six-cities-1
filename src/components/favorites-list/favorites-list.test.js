import React from 'react';
import renderer from 'react-test-renderer';
import FavoriteList from './favorites-list.jsx';

import {Router} from 'react-router-dom';
import history from '../../history';

import {createStore} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../../reducer/index';

const mockCities = new Set([
  `Amsterdam`,
  `Budapesht`,
  `Wroclaw`
]);

const mockOffers = [
  {
    isPremium: true,
    previewImage: ``,
    price: 222,
    isFavorite: true,
    id: 1,
    title: ``,
    type: ``,
    city: {
      name: `Wroclaw`
    }
  },
  {
    isPremium: true,
    previewImage: ``,
    price: 222,
    isFavorite: true,
    id: 2,
    title: ``,
    type: ``,
    city: {
      name: `Wroclaw`
    }
  },
];

it(`Test FavoriteCard Component`, () => {
  const tree = renderer
    .create(
        <Provider store={createStore(reducer)}>
          <Router history={history}>
            <FavoriteList cities={mockCities} offers={mockOffers}/>
          </Router>
        </Provider>
    )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
