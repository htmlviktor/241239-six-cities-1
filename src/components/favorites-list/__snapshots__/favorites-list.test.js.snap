// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FavoriteCard Component 1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    />
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Budapesht
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    />
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Wroclaw
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <article
        className="favorites__card place-card"
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            <img
              alt="Place image"
              className="place-card__image"
              height={110}
              src=""
              width={150}
            />
          </a>
        </div>
        <div
          className="favorites__card-info place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                222
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height={18}
                width={18}
              >
                <use
                  xlinkHref="#icon-bookmark--active"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/1"
              onClick={[Function]}
            >
              
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            
          </p>
        </div>
      </article>
      <article
        className="favorites__card place-card"
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/2"
            onClick={[Function]}
          >
            <img
              alt="Place image"
              className="place-card__image"
              height={110}
              src=""
              width={150}
            />
          </a>
        </div>
        <div
          className="favorites__card-info place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                222
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height={18}
                width={18}
              >
                <use
                  xlinkHref="#icon-bookmark--active"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/2"
              onClick={[Function]}
            >
              
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            
          </p>
        </div>
      </article>
    </div>
  </li>
</ul>
`;
